pkg/operator/bootstrapteardown/bootstrap_teardown_controller.go

```diff
@@ -165,17 +166,30 @@ func (c *BootstrapTeardownController) canRemoveEtcdBootstrap() (bool, bool, erro
 		return false, hasBootstrap, nil
 	}
 
-	if len(members) < 4 {
+	isUnsupportedUnsafeEtcd, err := c.isUnsupportedUnsafeEtcd()
+	if err != nil {
+		return false, hasBootstrap, err
+	}
+
+	switch {
+	case !isUnsupportedUnsafeEtcd && len(members) < 4:
 		return false, hasBootstrap, nil
+	case isUnsupportedUnsafeEtcd && len(members) < 2:
+		return false, hasBootstrap, nil
+	default:
+		// do nothing fall through
 	}
 
 	unhealthyMembers, err := c.etcdClient.UnhealthyMembers()
 	if err != nil {
 		return false, hasBootstrap, nil
 	}
 
-	// the etcd-bootstrap member is allowed to be unhealthy and can still be removed
+	// the etcd-bootstrap member is allowed to be unhealthy and can still be removed with exception of isUnsupportedUnsafeEtcd
 	switch {
+	// no member is allowed to be unhealthy if isUnsupportedUnsafeEtcd
+	case isUnsupportedUnsafeEtcd && len(unhealthyMembers) != 0:
```

Comment:
I don't see why this case is here.

Reply:
@deads2k if this is a two-member cluster etcd cluster one with etcd-bootstrap and other with master node, allowing an unhealthy member will lead to quorum loss and all subsequent `MemberRemove(etcd-bootstrap)` calls will fail

Reply:
read down, the default case covers this I think.

Reply:
for a two-node etcd cluster (single master openshift cluster) the workflow has to be:
1. Remove etcd-bootstrap member
2. destroy bootstrap

Reply:
hrm thinking again, not the default case but:
```
case len(unhealthyMembers) > 1:
```
will cover it because both the member will be unhealthy.

Reply:
I will update restoring the previous cases